samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -334,7 +334,7 @@ public void run(Duration timeout) {
     long t = System.currentTimeMillis();
     Map<SystemStreamPartition, List<IncomingMessageEnvelope>> output = new HashMap<>();
     HashSet<SystemStreamPartition> didNotReachEndOfStream = new HashSet<>(ssps);
-    while (System.currentTimeMillis() < t + timeout) {
+    while (System.currentTimeMillis() < t + timeout.toMillis()) {
```

Comment:
Minor: please change if it makes sense. 

We can do either of the two things here: 
1. `ExponentialSleepStrategy` util class in samza can be used here to simplify this while loop based timeout logic and have just the consumer fetch operation alone. 
2. Guava [TimeLimiter](https://google.github.io/guava/releases/23.0/api/docs/com/google/common/util/concurrent/TimeLimiter.html)  could be used to replace this retry logic.

Reply:
I have noted this down, will do in a cleanup patch